NOTES.md

```diff
@@ -1,3 +1,191 @@
+  * Added a `wait` command which waits for a table to be ready (#2259)
+  * Added `toJsonString` which converts a datum into a JSON string (#2513)
+  * Turned `map` into a variadic function for mapping over multiple sequences in parallel (#2574)
+  * Added a prefix version of `map` (#3321)
+  * Added an optional `squash` argument to the `changes` command which deafaults to `true`, letting the sever combine multiple changes to the same document (#2726, #3558)
+  * `min` and `max` will now use an index when passed the new `index` parameter (#2974, #2981)
+  * Improved changefeeds
+    * `changes` can now be placed after `between` and `filter` (#3232)
+    * `changes` can also be used on a few other commands for which there should be a comprehensive list here
+* Server
+  * Made buffered IO the default, added a `--direct-io` flag to enable direct IO and deprecated the `--no-direct-io` flag (#3391)
+* Python driver
+  * `rethinkdb export` now exports secondary index information and `rethinkdb import` re-creates those indexes unless it is given the `--no-secondary-indexes` flag (#3484)
+
+## Improvements ##
+
+* Web UI
+  * The web assets are now compiled into the `rethinkdb` executable (#1093)
+  * `getAll` queries on secondary indexes are now shown in the performance graph (#2379)
+  * Added `getField` to the autocompletion in the data explorer (#2624)
+  * Added support for changefeeds in the data explorer (#2643)
+  * Reduced the amount of data used and computations performed by the dashboard (#2786)
+  * Swapped the order of the "Load" and "Remove" buttons (#3414)
+  * The inaccessible database page was removed (#3491)
+  * Added a secondary index monitor to the dashboard (#3493)
+* Server
+  * Removed code used to support outdated tests (#1524)
+  * Improved implementation of erase range operations (#2034)
+  * Now use `jemalloc` by default instead of `tcmalloc`, which solves some apparent memory leaks (#2279)
+  * Replaced vector clocks with naïve timestamp-based conflict resolution (#2784)
+  * No longer complain when `stderr` can't be flushed (#2790)
+  * Replaced uses of the word "machine" to use "server" instead. The old `--machine-name` flag is now `--server-name` (#2877, #3254)
+  * The server now prints its own name on startup (#2992)
+  * Adjusted the formatting of log levels and no longer print `info:` lines to stderr (#3040)
+  * Added a `--no-default-bind` option that prevents the server from listening on loopback addresses (#3154)
+  * Lower the CPU load of an idle server (#3185)
+  * Now ignore the OS disk cache when calculating available memory (#3193)
+  * Now use `kqueue` instead of `poll` for events on OS X (#3403)
+  * Tables now become available faster and table metadata takes less space on disk (#3463)
+  * Queries that perform multiple HTTP requests now share the same cookies (#3469)
+  * Made resolving of ties in secondary indexes more consistent (#3550)
+  * New version notifications are added to the server logs (#3170)
```

Comment:
We should add that the server now "calls home" to check for newer versions. In fact, maybe there should be a more prominent notification, especially since Hacker News has been talking about privacy and surveillance so much lately.